src/BuiltInTools/dotnet-watch/LaunchBrowserFilter.cs

```diff
@@ -67,9 +67,12 @@ public async ValueTask ProcessAsync(DotNetWatchContext context, CancellationToke
                         context.Reporter.Verbose($"Refresh server running at {serverUrl}.");
                         context.ProcessSpec.EnvironmentVariables["ASPNETCORE_AUTO_RELOAD_WS_ENDPOINT"] = serverUrl;
 
+                        const string dotnetStartHooksName = "DOTNET_STARTUP_HOOKS";
                         var pathToMiddleware = Path.Combine(AppContext.BaseDirectory, "middleware", "Microsoft.AspNetCore.Watch.BrowserRefresh.dll");
-                        context.ProcessSpec.EnvironmentVariables["DOTNET_STARTUP_HOOKS"] = pathToMiddleware;
-                        context.ProcessSpec.EnvironmentVariables["ASPNETCORE_HOSTINGSTARTUPASSEMBLIES"] = "Microsoft.AspNetCore.Watch.BrowserRefresh";
+                        context.ProcessSpec.EnvironmentVariables[dotnetStartHooksName] = AddOrAppend(dotnetStartHooksName, pathToMiddleware);
```

Comment:
`DOTNET_STARTUP_HOOkS` has different separators based on the OS. https://github.com/dotnet/core-setup/blob/master/Documentation/design-docs/host-startup-hook.md#proposed-behavior. Can we make sure this is tested rather than rushing this change in at the last minute? If we're not planning to support 5.0 projects, there's no reason not to target the 6.0 (main / master) branch.

Reply:
Good catch; closing and reopening for master; will test both windows and unix before merging.